R/database-snomed.R

```diff
@@ -0,0 +1,536 @@
+#'   https://isd.digital.nhs.uk/trud/users/guest/filters/0/account/form
+#'
+#' - Obtain your **API key** from your TRUD profile.
+#'
+#' - Set your API key in your `.Renviron` file. You can open it using:
+#'   ```r
+#'   usethis::edit_r_environ()
+#'   ```
+#'   Then add a line like this (replace `<key>` with your API key):
+#'   ```
+#'   TRUD_API_KEY=<key>
+#'   ```
+#'
+#' - Verify that your key is set correctly:
+#'   ```r
+#'   Sys.getenv("TRUD_API_KEY")
+#'   ```
+#'
+#' ### Step 2: Subscribe to SNOMED CT Data
+#' - After logging into NHS TRUD, subscribe to the SNOMED CT items you need:
+#'   https://isd.digital.nhs.uk/trud/users/authenticated/filters/0/categories/1/items/1799/releases
+#'
+#' Ensure that your API key and subscriptions are active before running this function.
+#'
+#' ## Function Description
+#'
+#' `download_latestversion_of_snomed_item()` downloads and extracts the most recent release
+#' of a specified SNOMED CT item.
+#' By default, it retrieves the UK Clinical Edition Monolith (item 1799).
+#'
+#' @param item_number Numeric. The TRUD item number identifying the SNOMED CT UK data package.
+#'   Defaults to `1799`, which corresponds to the SNOMED CT UK Clinical Edition (Monolith).
+#'
+#' @param release_index Numeric. The index position of the TRUD metadata release to download.
+#'   Use `1` for the most recent release (e.g., `"uk_sct2mo_41.1.0_20251022000001Z.zip"`).
+#'   Valid indices typically range from `1` (latest) to `48` (oldest, e.g., `"Release 32.11.0"`).
+#'   **Note:** These index ranges are subject to change as new releases become available on TRUD.
+#'
+#' @param path_destination Character. Path to the directory where downloaded files
+#'   should be extracted. Defaults to the current working directory (`"."`).
+#'
+#' @return A named list containing:
+#' \describe{
+#'   \item{release_id}{The ID of the latest release downloaded.}
+#'   \item{file_path}{The full path to the downloaded ZIP file.}
+#'   \item{extracted_dir}{The directory where the files were extracted.}
+#' }
+#'
+#' @details
+#' The function uses the [`trud`](https://cran.r-project.org/package=trud) package to interact with the NHS TRUD API.
+#' Ensure you have installed and configured it correctly before use.
+#'
+#' @seealso
+#' [NHS TRUD Documentation](https://isd.digital.nhs.uk/trud/),
+#' [`trud::download_item()`](https://docs.ropensci.org/trud/)
+#'
+#' @examples
+#' \dontrun{
+#' # Download the default SNOMED CT UK Monolith
+#' result <- download_latestversion_of_snomed_item()
+#'
+#' # Download a specific earlier release and extract to a custom folder
+#' download_latestversion_of_snomed_item(
+#'   item_number = 1799,
+#'   release_index = 5,
+#'   path_destination = "data/snomed_releases"
+#' )
+#'
+#' # Check the extracted directory
+#' result$extracted_dir
+#' }
+#'
+#' @export
+download_latestversion_of_snomed_item <- function(
+  item_number = 1799,
+  release_index = 1,
+  path_destination = "."
```

Comment:
```suggestion
  destination_path = ".",
  item_number = 1799,
  release_index = 1
```

I would put the path as first argument, as it's the one mostly likely to be changed. So a user can do `download_snomed_item("/my/destination/path")` if they want to download the default item and release but change the path.